addon/components/mesh-manager.js

```diff
@@ -1,120 +1,80 @@
+  @service store;
+  @service intl;
+  @tracked query = '';
+  @tracked searchResults = [];
+  @tracked searchPage = 0;
+  @tracked hasMoreSearchResults = false;
 
-  sortedTerms: computed('terms.@each.name', function(){
-    var terms = this.get('terms');
-    if(!terms || terms.length === 0){
+  get sortedTerms(){
+    if(!this.args.terms || this.args.terms.length === 0){
       return [];
     }
-    return terms.sortBy('name');
-  }),
-  init(){
-    this._super(...arguments);
-    this.set('searchResults', []);
-    this.set('sortTerms', ['name']);
-  },
-  actions: {
-    search(query) {
-      this.set('searchReturned', false);
-      this.set('searching', true);
-      this.set('query', query);
-      var terms = this.get('terms');
-      this.get('store').query('mesh-descriptor', {
-        q: query,
-        limit: this.get('searchResultsPerPage') + 1
-      }).then(descriptors => {
-        const results = descriptors.map(function(descriptor){
-          return ProxiedDescriptors.create({
-            content: descriptor,
-            terms: terms
-          });
-        });
-        this.set('searchReturned', true);
-        this.set('searching', false);
-        this.set('searchPage', 1);
-        this.set('hasMoreSearchResults', (results.length > this.get('searchResultsPerPage')));
-        if (this.get('hasMoreSearchResults')) {
-          results.pop();
-        }
-        this.set('searchResults', results);
-      });
-    },
+    return this.args.terms.sortBy('name');
+  }
+  @restartableTask
+  *search(query) {
+    this.searching = true;
+    this.query = query;
 
-    clear() {
-      this.set('searchResults', []);
-      this.set('searchReturned', false);
-      this.set('searching', false);
-      this.set('searchPage', 0);
-      this.set('hasMoreSearchResults', false);
-      this.set('query', '');
-    },
-    add(term) {
-      const editable = this.get('editable');
-      if (editable) {
-        this.add(term.get('content'));
-      }
-    },
-    remove(term) {
-      const editable = this.get('editable');
-      if (editable) {
-        this.remove(term);
-      }
-    }
-  },
-  searchMore: task(function * () {
-    var terms = this.get('terms');
-    var query = this.get('query');
-    const descriptors = yield this.get('store').query('mesh-descriptor', {
+    const descriptors = yield this.store.query('mesh-descriptor', {
       q: query,
-      limit: this.get('searchResultsPerPage') + 1,
-      offset: this.get('searchPage') * this.get('searchResultsPerPage')
+      limit: SEARCH_RESULTS_PER_PAGE + 1
     });
-    const results = descriptors.map(function(descriptor){
-      return ProxiedDescriptors.create({
-        content: descriptor,
-        terms: terms
-      });
+
+    this.searching = false;
+    this.searchPage = 1;
+    this.hasMoreSearchResults = (descriptors.length > SEARCH_RESULTS_PER_PAGE);
+    if (this.hasMoreSearchResults) {
+      descriptors.pop();
```

Comment:
this blows up. descriptors is still and `DS.PromiseArray` object at this point, which subclasses `Ember.Array`, neither implement `pop()` . Please convert `descriptors` to a native JS array further up, using `toArray()`.

Reply:
Fixed, added tests even!